patches/components-sync-engine-syncer_proto_util.cc.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/components/sync/engine/syncer_proto_util.cc b/components/sync/engine/syncer_proto_util.cc
+index dfc4a2d32a81aab75ef4cf083a133029105a057e..f4b34d14ea69167d2952e50fab3f3d2c605f529e 100644
+--- a/components/sync/engine/syncer_proto_util.cc
++++ b/components/sync/engine/syncer_proto_util.cc
+@@ -440,6 +440,7 @@ SyncerError SyncerProtoUtil::PostClientToServerMessage(
+ 
+   SyncProtocolError sync_protocol_error =
+       GetProtocolErrorFromResponse(*response, cycle->context());
++  BRAVE_POST_CLIENT_TO_SERVER_MESSAGE
+ 
+   // Inform the delegate of the error we got.
+   cycle->delegate()->OnSyncProtocolError(sync_protocol_error);
```

Comment:
it looks like this goes directly into `SyncSchedulerImpl::OnSyncProtocolError`? we can set some var there that we then can use in `SyncSchedulerImpl::HandleError` to set our custom timer.
This way we should have to patch only 2 lines or even 1 (`OnSyncProtocolError` can possibly be define-patched).

Reply:
@goodov

>it looks like this goes directly into SyncSchedulerImpl::OnSyncProtocolError?


yes

Thanks, I understand the idea but I was able to make only partial proposition changes https://github.com/brave/brave-core/pull/13159/commits/b93336059a8da09543684ad6d3521994f5c7903f .
The problem is that `SyncSchedulerImpl::HandleFailure` is invoked after `SyncSchedulerImpl::OnSyncProtocolError`. So I can't reset the `wait_interval_` at 
`SyncSchedulerImpl::OnSyncProtocolError` because it would be set to `kExponentialBackoff` state again in `SyncSchedulerImpl::HandleFailure`.

Here how the stack trace look around these mentioned methods:
```
#2 0x56442ab78cae syncer::SyncSchedulerImpl::OnSyncProtocolError()
#3 0x56442ab85fcb syncer::SyncerProtoUtil::PostClientToServerMessage()
#4 0x56442ab8030e syncer::GetUpdatesProcessor::ExecuteDownloadUpdates()
#5 0x56442ab7fc01 syncer::GetUpdatesProcessor::DownloadUpdates()
#6 0x56442ab7e3ce syncer::Syncer::DownloadAndApplyUpdates()
#7 0x56442ab7ea99 syncer::Syncer::ConfigureSyncShare()
#8 0x56442ab7663b syncer::SyncSchedulerImpl::DoConfigurationSyncCycleJob()
#9 0x56442ab77fc8 syncer::SyncSchedulerImpl::TrySyncCycleJobImpl()

```
```
#2 0x56442ab758ee syncer::SyncSchedulerImpl::HandleFailure()
#3 0x56442ab766a9 syncer::SyncSchedulerImpl::DoConfigurationSyncCycleJob()
#4 0x56442ab77fc8 syncer::SyncSchedulerImpl::TrySyncCycleJobImpl()
```
Number of patched lines had been decreased :+1: !